### PR TITLE
Create and apply a storages superclass

### PR DESCRIPTION
**Pull request by TrustHenry**
Manage DB Instances and reduce code duplication.

This is the pre-requisite for further PR.

### CODE REVIEW

File: src/modules/storage/BlockStorage.ts [hunk 1]
```diff
@@ -14,34 +14,23 @@
...
      */
-    constructor (filename: string, callback?: any)
+    constructor (callback: any = null)
     {
-        this.db = new sqlite.Database(filename,
-            sqlite.OPEN_CREATE | sqlite.OPEN_READWRITE |
-            sqlite.OPEN_SHAREDCACHE);
-        this.db.configure("busyTimeout", 1000);
-        this.createTable(callback);
+        super(undefined, callback);
```
Comment (AndrejMitrovic) [inline src/modules/storage/BlockStorage.ts]: Am I missing something here? You are calling the base constructor with `undefined`?

File: src/modules/storage/Storages.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Constructor
+     * @param filename Valid values are filenames,
+     * ":memory:" for an anonymous in-memory database and
+     * an empty string for an anonymous disk-based database
+     * @param callback : If provided, this function will be called when
+     * the database was opened successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    constructor (filename : string = ".cache\ledger", callback : any = null)
```
Comment (AndrejMitrovic) [inline src/modules/storage/Storages.ts]: filename should not be a default parameter if you want this class to be reusable. What if you forget to set the filename? then you end up overwriting the ledger storage by mistake.

File: src/modules/storage/BlockStorage.ts (hunk 1 above)
Comment (TrustHenry) [reply src/modules/storage/BlockStorage.ts]: It seems unnecessary to me, too.
DB filename can be fixed.

File: src/modules/storage/Storages.ts (hunk 2 above)
Comment (TrustHenry) [reply src/modules/storage/Storages.ts]: Thank you reviews.
^ Fixed the DB filename.

File: src/modules/storage/Storages.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * @param filename Valid values are filenames,
+     * ":memory:" for an anonymous in-memory database and
+     * an empty string for an anonymous disk-based database
+     * @param callback : If provided, this function will be called when
+     * the database was opened successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    constructor (callback: any = null)
+    {
+        let filename: string = ".cache";
```
Comment (AndrejMitrovic) [inline src/modules/storage/Storages.ts:36]: I still don't understand what you're trying to accomplish.

You will have multiple instances of `Storages` all writing to the same file?

Comment (TrustHenry) [reply src/modules/storage/Storages.ts:36]: Yes, multiple Storages are managed in a single file.
I'd like to join a number of tables and use them.

Comment (TrustHenry) [reply src/modules/storage/Storages.ts:36]: Is it because the name storage is strange?

Comment (AndrejMitrovic) [reply src/modules/storage/Storages.ts:36]: Then why do you need multiple instances of `sqlite.Database`? If you will only use one file then you only need one database handle.

Comment (MichaelKim20) [reply src/modules/storage/Storages.ts:36]: I think this is what @AndrejMitrovic  wants.
I think this is good too.

```suggestion
    constructor (filename: string, callback?: any)
```

Comment (AndrejMitrovic) [reply src/modules/storage/Storages.ts:36]: Yes that is what I thought Henry tried to do at first. But according to him he wants a single file for all the tables..

File: src/modules/storage/Storages.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Constructor
+     * @param filename Valid values are filenames,
+     * ":memory:" for an anonymous in-memory database and
+     * an empty string for an anonymous disk-based database
+     * @param callback : If provided, this function will be called when
+     * the database was opened successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    constructor (callback?: any)
```
Comment (MichaelKim20) [inline src/modules/storage/Storages.ts:34]:
```suggestion
    constructor (filename: string, callback?: any)
```

File: src/modules/storage/Storages.ts (hunk 3 above)
Comment (AndrejMitrovic) [reply src/modules/storage/Storages.ts:36]: I'll leave design decisions to you two. It was just a comment.

Comment (TrustHenry) [reply src/modules/storage/Storages.ts:36]: Maybe @AndrejMitrovic  talking about a single-ton design pattern. Is that right?

Comment (TrustHenry) [reply src/modules/storage/Storages.ts:36]: Then it should get a DB Instance through `getDbInstancea`
In subsequent operations, you can manage instances as one after creating superclass in `Stoa.ts`

This PR is still applied only to the test class.

File: src/modules/storage/Storages.ts (hunk 4 above)
Comment (TrustHenry) [reply src/modules/storage/Storages.ts:34]: I don't think it's necessary yet.

**Review by MichaelKim20 — APPROVED** [review]
LGTM

**Pull request merged**